### ISSUE TITLE
Nativewind 4 stable release

### ISSUE THREAD

**Issue by raghav-reglobe** [nativewind/nativewind#976]
Dear @marklawlor, I was using tailwind in ReactJs and My main Application was in **Flutter**.
One day I decided to migrate my main application to react-native(new architecture) by using ReactJs code base along with **Nativewind@4**.
After investing 4 months in migration and optimization , I am unable to release my first build on production due to poor performance and it's not due to **Nativewind** alone, I am working on it.

**It would be very helpful if you can share any timeline of stable release.**
Is it anywhere related to **virtual-modules** branch you are working on?

**Comment by venables** [comment]
@raghav-reglobe -- the pinned issue in this repository mentions they intend to release mid-to-late July.

https://github.com/nativewind/nativewind/issues/600#issuecomment-2192912430

**Comment by umardev500** [comment]
I am waiting for v4

**Comment by prince272** [comment]
Likewise!

**Comment by ducpt-bili** [comment]
Hi @raghav-reglobe , i am really curious about what you said aboud poor performance. Do you mean your RN app with tech stack: tailwind v4, new arch ... and not have the better performance compare to the flutter? Thanks.

**Comment by raghav-reglobe** [comment]
Hi @ducpt-bili thanks for your question, yes my current tech stack is flutter, it not about better performance, I am struggling to match it's performance.

**Comment by ducpt-bili** [comment]
hi @raghav-reglobe , so you mean RN stack like above it as far as good as current flutter project right?

**Comment by raghav-reglobe** [comment]
@danstepanov, May I ask for an update of the timeline you have shared in https://github.com/nativewind/nativewind/issues/600#issuecomment-2192912430

**Comment by danstepanov** [comment]
TL;DR: Mark and I are working through a bug with react-native-reanimated to get the remaining test suites passing (currently 97/105 tests are passing). Aiming to have a minor version out end of this week assuming we resolve the issues.

I post more frequent updates [on Twitter](https://x.com/nativewindcss/status/1819507491874783407). 

Re whatever the hell you're talking about @shawnmclean, we don't **have** to do anything. We do this work for free and alongside the work that pays us. The branch we're working off of is the `virtual-modules` branch, contributions are always welcome. I'll continue to communicate re our progress/timeline as it becomes clear to me.

**Comment by danstepanov** [comment]
https://x.com/nativewindcss/status/1821281884606992471

**Comment by danstepanov** [comment]
> @danstepanov thanks for the update.
> 
> For those who do not use Twitter, what we see is what happens within GitHub.
> 
> Also, nobody has to do anything in this world, this is one of those responses given by folks who run communities and other OSS projects.
> 
> However, regardless of free work or not, one must communicate if your work impacts others. If there is no communication, or visibility into any sort of work, then it is either perceived that nothing is happening or whatever is happening is not in the benefit of those who you want your work to impact.
> 
> Looking at the Twitter(which one must have an account to see the tweets), thank you for your contributions.
> 
> The point of contacts for me and others is GitHub and the website. If Twitter and discord are new channels, that's also fine and should have something somewhere on existing channels for us to flow into those.
> 
> I do not wish to offend, I am simply pointing out communication and transparency flaws and improvements.
> 
> One cannot help if one does not see where help is needed. Especially if the 2 maintainers are sorting out bugs in communication channels that we are unable to see.
> 
> Now regarding my statement about other UI component library, I usually look up the maintainers of libraries I use to see what their incentive structures are if I smell something is off.
> 
> From your profile and the maintainer of rn-reusables, I see both of you working on https://nativewindui.com/. First glance on the site I see a price, even though I found it through github.
> 
> And if that's the incentive structure for maintaining nativewind, that's perfectly fine with me.
> 
> But as I said, it's the communication and transparency issues within the communication channels that we currently use.
> 
> I'm sure many of us here are willing to help, myself included.
> 
> Again, thanks for the contributions and nobody has to do anything 😉
> 
> PS. I'm still in between gluestack, rn-reusables and now I see nativewindui, which means it may be in my best interest to go your route since you're maintaining this thing.

You're right re people not having Twitter, I'll post updates here as well. Thanks for the feedback. I'll have contribution docs up shortly as well.

As an additional update re NativeWind v4.1, all tests are now passing!

**Comment by raghav-reglobe** [comment]
@danstepanov thanks for the update, if **all tests are passing**. Please share us for testing the performance or the tentative timeline to release for everyone.

**Comment by danstepanov** [comment]
📢 v4.1 Update

Just one more test left to fix…

**Comment by danstepanov** [comment]
📢 v4.1 Update

🎉 All tests are passing! 🎉

Last thing to do is to fix animation switching. Animations work on their own; however, switching and/or changing them causes things to not work/crash.

We’ll start adding people to the Discord today for testing this week.

https://x.com/nativewindcss/status/1823061543283720631?s=46&t=ktI223rpD8Wzr88UqL4JQg

**Comment by danstepanov** [comment]
📢 v4.1 Update

Removing one polyfill today/tomorrow. Then just writing up some release notes. 

Canary release this week!

Discord is live
https://discord.gg/MqrnNWNj

https://x.com/nativewindcss/status/1826134088941543928?s=46&t=ktI223rpD8Wzr88UqL4JQg

**Comment by danharten-sovtech** [comment]
SO relieved I found this thread! I was extremely worried I wouldn't be able to use this in Production with no clear updates after spending many months on my codebase. I was about to take the hit and make a hard switch. Not anymore :)

Thanks for posting updates here now @danstepanov 🚀 heads up I am sure there are others in a similar position having a hard time finding this 👀 

Cheers for your hard work 🍻

**Comment by danstepanov** [comment]
Mark deserves a lot of the credit here. I'm still getting my bearings but really appreciate the kind words ❤️

As far as v4.1 update, we're aiming to merge it to `main` this upcoming Monday 🙂 

There's a lot more ahead so stay tuned!

**Comment by danstepanov** [comment]
https://github.com/nativewind/nativewind/issues/600#issuecomment-2322764107
